components/properties/src/props.rs

```diff
@@ -624,3 +626,75 @@ impl SentenceBreak {
     pub const LF: SentenceBreak = SentenceBreak(13); // name="LF"
     pub const SContinue: SentenceBreak = SentenceBreak(14); // name="SC"
 }
+
+/// Enumerated property Canonical_Combining_Class.
+/// See UAX #15:
+/// <https://www.unicode.org/reports/tr15/>.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[repr(transparent)]
+pub struct CanonicalCombiningClass(pub u8);
+
+// These constants come from the VALUES_ccc table in pnames_data.h from ICU4C.
```

Comment:
I would refer to the ccc value aliases in https://www.unicode.org/Public/UCD/latest/ucd/PropertyValueAliases.txt rather than to an internal header file in an offline builder tool.

Moreover, if the trie lookup returns an integer, I don't know if there is any use case for these named constants in ICU4X. ICU4C/J haven't needed such enums.

Of course, someone somewhere needs to be able to map from a value alias string to a ccc value (integer), for regex pattern parsing, but that should not require defining an enum with named value constants.

Reply:
I regenerated the constants from PropertyValueAliases.txt. The result was the same, except for an extra comment "RESERVED".

Reply:
I have dropped the word "enumerated" from the documentation..

Instead of making the return value `u8` directly, I derived `Ord` and `PartialOrd` for `CanonicalCombiningClass`. This way, we both have strong typing (this is a `CanonicalCombiningClass` rather than a generic `u8`), you can compare for equality with a constant of interest (likely only `NotReordered`), and you can sort directly without having to explicitly extract the `u8`. However, if, for some reason, you need the wrapped `u8`, you can still get it.

Reply:
FYI for those not steeped in Rust: 

This ["`newtype` idiom"](https://doc.rust-lang.org/rust-by-example/generics/new_types.html) in Rust is just a wrapper struct around a primitive type. IIUC, you get some [pros that outweigh the cons](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html) with the help of the Rust language/compiler making this a "zero-cost abstraction".

The whole discussion in ICU4X previously about whether to use a [Rust enum vs. newtype](https://docs.google.com/document/d/1Ocsmakc9ydXaJ-3ntCd25H3wnCHTRe0S1sRUzJ0iqvc/edit#heading=h.pgya1lt8lqxt) to represent Unicode properties came down to whether the set of valid property values for a property would increase over time or not. Rust enums expect a closed set of values, and the language supports that design elsewhere (ex: the Rust language allows match statements on enums (kind of like Java 12 switch expressions) to omit the `default` case when all known cases are accounted for in the case branch conditions). For the time being, only `GeneralCategory` is a Rust enum, and all other Unicode properties are represented as newtypes. The primitive type that is chosen to be wrapped by the newtype that we define for a Unicode property is usually informed by whether the `CodePointTrie` of ICU exported data gives us a `u8`, `u16`, or `u32`.

Also, in Rust, once you have this newtype, then you can start doing other things that are specific to this new thing you've defined. [`Ord`](https://doc.rust-lang.org/std/cmp/trait.Ord.html) and [`PartialOrd`](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html) are like Java [`Comparable`](https://docs.oracle.com/javase/8/docs/api/java/lang/Comparable.html). ([why both Ord & PartialOrd? -> sort-order-able vs. total-order-able](https://stackoverflow.com/questions/59315296/why-does-rust-need-the-partialord-trait-in-addition-to-the-ord-trait#comment104832175_59315296)). Because Rust traits are like Java 8 interfaces, they can have default implementations. To "derive a trait [on a struct]" is to implement a trait using its default implementation, done via a `#[derive(TraitName)]` directive on top of the struct/enum definition. As Henri pointed out, [deriving Ord](https://doc.rust-lang.org/std/cmp/trait.Ord.html#derivable) and [deriving PartialOrd](https://doc.rust-lang.org/std/cmp/trait.PartialOrd.html#derivable) on the newtype for Canonical_Combining_Class should work correctly as desired so that you can avoid having to pull out the wrapped values in sorting & other scenarios.